### DISCUSSION TITLE
OsmAnd for paragliding (free flying sports) with 3D tracks

### DISCUSSION THREAD

**Discussion by elgandoz** [osmandapp/OsmAnd#18035] [Ideas]
OsmAnd is a great tool for planning paragliding adventures, thanks to its many features.
I use it in a dedicated combination of elements, which includes Paragliding (free flying) POI, 3D elevation, hike routes with mountain peaks, hypsometric raster Underlay with contour lines and [OpenFlightMaps](https://www.openflightmaps.org/)/[Thermal KK7](https://thermal.kk7.ch/) overlays, but the options are endless.
The app provides a great way to plan a hike&fly or even regular Cross Country flights [aka XCs], and even the 8 time RedBull X-Alps and PWC champion [Chrigel Maurer uses it to plan his legendary races](https://www.reddit.com/r/OsmAnd/comments/kpuuq5/chrigel_maurer_3x_paragliding_world_cup_champion/).

A small set of improvements could make it even better for this purpose.
Here's a suggestion list, which I will then convert in feat. requests tickets.

- [x] [Paragliding Icon for profiles](https://github.com/osmandapp/OsmAnd/issues/18041)
- [x] [3D tracklogs for GPX](https://github.com/osmandapp/OsmAnd/issues/18040)
- [ ] [bonus] Reading of [IGC files](https://en.wikipedia.org/wiki/IGC_(file_format)) natively
- [ ] [optional] Default inclusion of [Thermal KK7](https://thermal.kk7.ch/)

These are features that could also be used by a whole other set of user aside from paragliding, like other free flying and general aviation pilots, but also drone pilots and whoever may need to display a 3-dimensional track.

If anyone has other related ideas, please share it here.

![Screenshot_20230828-163500](https://github.com/osmandapp/OsmAnd/assets/5859926/1c944f38-dc21-468a-9d1e-4e9e530053c1)

**Comment by vshcherb** [comment]
Sounds cool, though some of features like Thermal KK7 will be very tricky to implement for generic purpose app
```
Thermal maps show the probablity to find thermal uplift at a certain geographical location using a paraglider. The maps are soly based on real flights
```

**Comment by elgandoz** [comment]
KK7 layers ("thermalmaps" and "skyways") are already publicly available as raster tiles. The screenshot above is using it (with my own API key - the blue/green/red spots).
Other 3rd party apps are already using it, it would be only matter of getting the permission and including the layers in the default list of available raster tiles for the "Online maps" plugin.

More info: https://thermal.kk7.ch/#code

**Reply by elgandoz** [reply]
An alternative would be allowing users to download an OsmAnd-ready SQLiteDB file from the Thermal KK7 website.

**Comment by pierre-quelin** [comment]
Some other ideas
- [ ] Retrieve airspaces maps like XCTrack (https://airspace.xcontest.org/...).
- [ ] Displaying the maximum theoretical glide distance on the map in the same way as the https://www.spotair.mobi/ website with "Hike & Fly" button, either on demand or live, would be great too.
- [ ] Some cool widgets, like an artificial horizon so you never lose your bearings in the clouds or calculating wind speed and direction.

I'm sure that the functions will be extended, because this application is the best I've ever seen. It's been getting better and better since I started using it many years ago. Thanks to all the developers who do such a fantastic job.

**Comment by lenalebt** [comment]
Hey,
did you go any step in that direction? I had roughly the same idea and came here because of that. I thought maybe one can build this as a plugin? Not sure whether plugins can do all the things we'd need to have for that...

Other features I'd love to have, roughly:
- show nearby official landing spots
- obstacles
- train / bus stations shown at the same time, with different colors

Cheers,
Lena
